mitosheet/src/components/taskpanes/Graph/GraphSetupTab.tsx

```diff
@@ -0,0 +1,337 @@
+
+
+/* 
+    The graph setup tab where the user creates the structure of the graph by 
+    selecting data
+*/
+function GraphSetupTab(
+    props: {
+        uiState: UIState;
+        mitoAPI: MitoAPI;
+        sheetDataArray: SheetData[];
+        graphParams: GraphParams
+        dfNames: string[];
+        columnDtypesMap: Record<string, string>;
+        columnIDsMapArray: ColumnIDsMap[],
+        setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+        setGraphParams: React.Dispatch<React.SetStateAction<GraphParams>>;
+        setGraphUpdatedNumber: React.Dispatch<React.SetStateAction<number>>;
+    }): JSX.Element {
+
+    const graphSheetIndex = props.graphParams.graphCreation.sheet_index
+
+    // Toggles the safety filter component of the graph params
+    const toggleSafetyFilter = (): void => {
+        const newSafetyFilter = !props.graphParams.graphPreprocessing.safety_filter_turned_on_by_user
+
+        props.setGraphParams(prevGraphParams => {
+            const copyPrevGraphParams = {...prevGraphParams}
+            return {
+                ...copyPrevGraphParams,
+                graphPreprocessing: {
+                    safety_filter_turned_on_by_user: newSafetyFilter
+                }
+            }
+        })
+        props.setGraphUpdatedNumber((old) => old + 1);
+    }
+
+    /* 
+        Function responsible for updating the selected column headers for each axis. 
+        Set the columnHeader at the index of the graphAxis selected columns array.
+    
+        To remove a column, leave the columnHeader empty.
+    */
+    const updateAxisData = (graphAxis: GraphAxisType, index: number, columnID?: ColumnID) => {
+        // Get the current axis data
+        let axisColumnIDs: ColumnID[] = []
+        if (graphAxis === GraphAxisType.X_AXIS) {
+            axisColumnIDs = props.graphParams.graphCreation.x_axis_column_ids
+        } else {
+            axisColumnIDs = props.graphParams.graphCreation.y_axis_column_ids
+        }
+
+        // Make a copy of the column headers before editing them
+        const axisColumnIDsCopy = [...axisColumnIDs]
+
+        if (columnID === undefined) {
+            axisColumnIDsCopy.splice(index, 1)
+        } else {
+            axisColumnIDsCopy[index] = columnID
+        }
+
+        // Update the axis data
+        if (graphAxis === GraphAxisType.X_AXIS) {
+            props.setGraphParams(prevGraphParams => {
+                const copyPrevGraphParams = {...prevGraphParams}
+                return {
+                    ...copyPrevGraphParams,
+                    graphCreation: {
+                        ...copyPrevGraphParams.graphCreation, 
+                        x_axis_column_ids: axisColumnIDsCopy
+                    }
+                }
+            })
+        } else {
+            props.setGraphParams(prevGraphParams => {
+                const copyPrevGraphParams = {...prevGraphParams}
```

Comment:
A double copy here. But also, it's not a nested copy... so not sure what the deal is here. 

Maybe we want to change both of these to:

```
props.setGraphParams(prevGraphparams => {
    return {
        graphCreation: {
             ... prevGraphparams,
             y_axis_column_ids: axisColumnIDsCopy
        }, 
        graphStyling: {
             ....prevGraphParams.graphStyling
        },
        ....
    }
}
```
Aka, given that I don't think it's making deep copies of the object, I think we actually want to go ahead and make them ourselves in the nested objects. 

The other option here (which is actually better, and what we should do!) is to create a deep copy of this object just with:

```
props.setGraphParams(prevGraphparams => {
    const graphParamsCopy = JSON.parse(JSON.stringify(graphParamsCopy)); 
    return {
        ...graphParamsCopy,
        graphCreation: {
             ... prevGraphparams,
             y_axis_column_ids: axisColumnIDsCopy
        }, 
    }
}
```

This is best for a few reasons:
1. It will probably fix the key issues I commented about already. This failure to make new objects is probably where that issue is occurring. 
2. It will scale as we continue to nest objects (e.g. for styling!). 

Note here: you need to make this change in a few places (namely, any place where we set graph params).